docs(readme): update readme to reflect this repo (#20)
mun-lang/mun: README.md

  machine code that can be natively executed on any target platform. LLVM is used for compilation
  and optimization, guaranteeing the best possible performance. Hot reloading does introduce a
  slight runtime overhead, but it can be disabled for production builds to ensure the best possible
  runtime performance.

- **Cross compilation** - The Mun compiler is able to compile to all supported target platforms from
  any supported compiler platform.

- **Powerful IDE integration** *not implemented yet* - The Mun language and compiler framework are
  designed to support source code queries, allowing for powerful IDE integrations such as code
  completion and refactoring tools.

## Examples

```mun
fn main() {
    let sum = add(a, b);

    // Comments: Mun natively supports bool, float, and int
    let is_true = true;
    let var: float = 0.5;
    
}

// The order of function definitions doesn't matter
fn add(a: int, b: int): int {
    a + b
}
```

## Documentation

[The Mun Programming Language Book](https://docs.mun-lang.org/)

## Building from Source

### Installing dependencies

Make sure you have the following dependencies installed on you machine:

#### Rust

> Install the last stable Rust version [using for instance

Reviewer: ```suggestion
Install the latest stable version of Rust, [e.g. using
```

rustup](https://www.rust-lang.org/tools/install). 

#### LLVM

> Mun targets LLVM 7.1.0. Installing LLVM can be a pain and is very platform dependant. The following

Reviewer: ```suggestion
Mun targets LLVM 7.1.0. Installing LLVM is platform dependant and as such can be a pain. The
```

> steps are how we install LLVM on [our CI

Reviewer: ```suggestion
following steps are how we install LLVM on [our CI
```

runners](https://github.com/mun-lang/mun/blob/master/ci/azure-install-llvm.yml):

> * ***nix**: Recent versions of *nix have binary versions of LLVM available. e.g.:

Reviewer: ```suggestion
* ***nix**: Package managers of recent *nix distros can install binary versions of LLVM, e.g.:
```

  ```bash
  # Ubuntu 18.04
  sudo apt install llvm-7 llvm-7-* lld-7
  ```

> * **macOS**: [Brew](https://brew.sh/) contains a binary distribution of LLVM 7.1.0. However because

Reviewer: ```suggestion
* **macOS**: [Brew](https://brew.sh/) contains a binary distribution of LLVM 7.1.0. However, as it's
```

>   its not the latest version its not added to the path. We are using

Reviewer: ```suggestion
  not the latest version, it won't be added to the path. We are using
```

>   [llvm-sys](https://crates.io/crates/llvm-sys), another option is to export the

Reviewer: ```suggestion
  [llvm-sys](https://crates.io/crates/llvm-sys) to manage version, but another option is to export the
```

>   `LLVM_SYS_70_PREFIX` which will not clutter you `PATH`. To install:

Reviewer: ```suggestion
  `LLVM_SYS_70_PREFIX` variable, which will not clutter your `PATH`. To install:
```

  ```bash
  brew install llvm@7
  # Export LLVM_SYS_PREFIX to not clubber PATH
  export LLVM_SYS_PREFIX=$(brew --prefix llvm@7)
  ```

> * **windows**: There are binary distrubutions available for Windows from the LLVM website but they

Reviewer: ```suggestion
* **windows**: Binary distrubutions are available for Windows on the LLVM website, but they
```

>   do not contain a number of libraries that are required by Mun. To not have to go to the trouble of

Reviewer: ```suggestion
  do not contain a number of libraries that are required by Mun. To avoid having to go to the trouble of
```

>   compiling LLVM yourself we created a

Reviewer: ```suggestion
  compiling LLVM yourself, we created a
```

>   [repository](https://github.com/mun-lang/llvm-package-windows) with a script that compiles the

Reviewer: ```suggestion
  [repository](https://github.com/mun-lang/llvm-package-windows) that automatically compiles the
```

>   binaries required. It also contains a

Reviewer: ```suggestion
 required binaries. It also contains a
```

  [release](https://github.com/mun-lang/llvm-package-windows/releases/download/v7.1.0/llvm-7.1.0-windows-x64-msvc15.7z)

>   that you can download and extract on you machine. Once downloaded and extracted, add the `<extract_dir>/bin`

Reviewer: ```suggestion
  that you can download and extract to your machine. Once downloaded and extracted, add the `<extract_dir>/bin`
```
